src/diffusers/models/transformers/transformer_wan.py

```diff
@@ -114,9 +115,9 @@ def __init__(self, in_features: int, out_features: int):
         self.norm2 = nn.LayerNorm(out_features)
 
     def forward(self, encoder_hidden_states_image: torch.Tensor) -> torch.Tensor:
-        hidden_states = self.norm1(encoder_hidden_states_image)
+        hidden_states = self.norm1(encoder_hidden_states_image.float()).type_as(encoder_hidden_states_image)
```

Comment:
we cannot always assume `self.norm1` is always in `float32` here. i.e for layers in `_keep_in_fp32_modules`, user can still cast it into a different dtype if they do something like  `model.to(torch.float16)`

maybe change to FP32LayerNorm ~